enterprise/backend/test/metabase_enterprise/tenants/api_test.clj

```diff
@@ -0,0 +1,638 @@
+                                                             :location (collection/children-location tenant-coll)
+                                                             :archived true}]
+              (testing "deleting the collection is allowed"
+                (mt/user-http-request :crowberto :delete 200 (str "collection/" child-id)))
+
+              (testing "child collection still exists"
+                (is (not (t2/exists? :model/Collection :id child-id)))))))))))
+
+(deftest cannot-move-tenant-root-collection-test
+  (testing "Cannot move tenant-specific-root-collection to a different parent"
+    (mt/with-premium-features #{:tenants}
+      (mt/with-temporary-setting-values [use-tenants true]
+        (mt/with-temp [:model/Tenant {tenant-collection-id :tenant_collection_id} {:name "Tenant Test" :slug "test"}
+                       :model/Collection {target-id :id} {:name "Target Collection"
+                                                          :location "/"
+                                                          :namespace :tenant-specific}]
+          (testing "attempting to move tenant root collection returns 400"
+            (mt/user-http-request :crowberto :put 400 (str "collection/" tenant-collection-id)
+                                  {:parent_id target-id}))
+
+          (testing "tenant root collection location remains at root"
+            (is (= "/" (t2/select-one-fn :location :model/Collection :id tenant-collection-id)))))))))
+
+(deftest can-move-tenant-descendants-within-tenant-namespace-test
+  (testing "Can move tenant collection descendants within the same tenant namespace"
+    (mt/with-premium-features #{:tenants}
+      (mt/with-temporary-setting-values [use-tenants true]
+        (mt/with-temp [:model/Tenant {tenant-collection-id :tenant_collection_id} {:name "Tenant Test" :slug "test"}]
+          (let [tenant-coll (t2/select-one :model/Collection :id tenant-collection-id)]
+            (mt/with-temp [:model/Collection {child-a-id :id} {:name "Child A"
+                                                               :namespace :tenant-specific
+                                                               :location (collection/children-location tenant-coll)}
+                           :model/Collection {child-b-id :id} {:name "Child B"
+                                                               :namespace :tenant-specific
+                                                               :location (collection/children-location tenant-coll)}]
+              (testing "can move child B under child A"
+                (mt/user-http-request :crowberto :put 200 (str "collection/" child-b-id)
+                                      {:parent_id child-a-id})
+
+                (is (= (str "/" tenant-collection-id "/" child-a-id "/")
+                       (t2/select-one-fn :location :model/Collection :id child-b-id)))))))))))
+
+(deftest cannot-move-regular-collection-into-tenant-namespace-test
```

Comment:
Because these aren't testing the tenants api could they got in another test ns as a sibling to this one (like collections_test.clj)